compiler/rustc_typeck/src/check/demand.rs

```diff
@@ -752,8 +752,8 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             }
         }
 
-        let msg = format!("you can convert an `{}` to `{}`", checked_ty, expected_ty);
-        let cast_msg = format!("you can cast an `{} to `{}`", checked_ty, expected_ty);
+        let msg = format!("you can convert {} `{}` to `{}`", checked_ty.kind().article(), checked_ty, expected_ty);
+        let cast_msg = format!("you can cast {} `{} to `{}`", checked_ty.kind().article(), checked_ty, expected_ty);
```

Comment:
I think it might be better as:

```suggestion
        let msg = format!("you can convert {} `{}` to {} `{}`", checked_ty.kind().article(), checked_ty, expected_ty.kind().article(), expected_ty);
        let cast_msg = format!("you can cast {} `{} to {} `{}`", checked_ty.kind().article(), checked_ty, expected_ty.kind().article(), expected_ty);
```

but I'm not sure. What do you think?

Reply:
I think it would look like:

```
help: you can convert a `usize` to an `isize`
help: you can cast a `u8` to a `u16`
```

Reply:
I think that change would be reasonable. You can go ahead and do it. Don't forget to reformat the code (and run `./x.py fmt`).